qcodes/tests/dataset/test_measurement_context_manager.py

```diff
@@ -1473,15 +1493,15 @@ def test_datasaver_multi_parameters_scalar(channel_array_instrument):
     param = channel_array_instrument.A.dummy_scalar_multi_parameter
     meas.register_parameter(param)
     assert len(meas.parameters) == len(param.shapes)
-    assert tuple(meas.parameters.keys()) == tuple(param.names)
+    assert set(meas.parameters.keys()) == set(param.names)
 
     with meas.run() as datasaver:
         datasaver.add_result((param, param()))
 
-    assert datasaver.points_written == 1
+    assert datasaver.points_written == 2
     ds = load_by_id(datasaver.run_id)
-    assert ds.get_data('thisparam') == [[0]]
-    assert ds.get_data('thatparam') == [[1]]
+    assert [d for d in ds.get_data('thisparam') if d[0] is not None] == [[0]]
+    assert [d for d in ds.get_data('thatparam') if d[0] is not None] == [[1]]
```

Comment:
so now "points_written == 2" and some rows have now "None" values. Just to keep myself up to date, what are the reasons for these to change?